backend/logic/order/index.js

```diff
@@ -194,6 +194,30 @@ async function processNewOrder({
   const order = await Order.create(orderObj)
   log.info(`Saved order ${order.fqId} to DB.`)
 
+  if (shopConfig.inventory) {
+    const cartItems = get(data, 'items', [])
+    const dbProducts = await Product.findOne({
+      shopId: shop.id,
+      productId: cartItems.map((item) => item.product)
+    })
+
+    for (const product of dbProducts) {
+      const item = cartItems.find((item) => item.product === product.productId)
+      const variantId = get(item, 'variant')
+
+      if (variantId) {
+        product.update({
+          stockLeft: product.stockLeft - item.quantity,
```

Comment:
What if there is not enough stock? The Front-end does some checks but in case of multiple buyers sending their order in a short timespan, shouldn't the back-end check?

It may not be easy to handle with our current asynchronous architecture...
I'm actually a similar problem while looking at the issue related to over-use of a discount code on the Injective shop. The back-end did not refuse those orders where the discount was over-used. 

But let's think about possible solutions...

Reply:
I added some checks for it but it justs add an error message to the orderObj instead of throwing. Because if we throw and don't create the Order, there will be no track of the order in Dshop and sellers have to search through their PayPal or Stripe accounts to process the refund